Makefile

```diff
@@ -1,56 +1,99 @@
-	@echo "lint - check style with flake8"
-	@echo "test - run tests quickly with the default Python"
-	@echo "test-all - run tests on every Python version with tox"
-	@echo "coverage - check code coverage quickly with the default Python"
-	@echo "docs - generate Sphinx HTML documentation, including API docs"
-	@echo "release - package and upload a release"
-	@echo "sdist - package"
-
-clean: clean-build clean-pyc
-
-clean-build:
+	@python -c "$$PRINT_HELP_PYSCRIPT" < $(MAKEFILE_LIST)
+
+clean: clean-build clean-pyc clean-test ## remove all build, test, coverage and Python artifacts
+
+clean-build: ## remove build artifacts
 	rm -fr build/
 	rm -fr dist/
-	rm -fr *.egg-info
+	rm -fr .eggs/
+	find . -name '*.egg-info' -exec rm -fr {} +
+	find . -name '*.egg' -exec rm -f {} +
+
+clean-docs: ## remove docs artifacts
+	rm -f docs/apidoc/xncml*.rst
+	rm -f docs/apidoc/modules.rst
+	$(MAKE) -C docs clean
 
-clean-pyc:
+clean-pyc: ## remove Python file artifacts
 	find . -name '*.pyc' -exec rm -f {} +
 	find . -name '*.pyo' -exec rm -f {} +
 	find . -name '*~' -exec rm -f {} +
+	find . -name '__pycache__' -exec rm -fr {} +
+
+clean-test: ## remove test and coverage artifacts
+	rm -fr .tox/
+	rm -f .coverage
+	rm -fr htmlcov/
+	rm -fr .pytest_cache
+
+lint/flake8: ## check style with flake8
+	ruff xncml tests
+	flake8 --config=.flake8 xncml tests
 
-lint:
-	flake8 xncml test
+lint: lint/flake8 ## check style
 
-test:
-	py.test
+test: ## run tests quickly with the default Python
+	python -m pytest
 
-test-all:
+test-all: ## run tests on every Python version with tox
 	tox
 
-coverage:
-	coverage run --source xncml setup.py test
+coverage: ## check code coverage quickly with the default Python
+	coverage run --source xncml -m pytest
 	coverage report -m
 	coverage html
-	open htmlcov/index.html
+	$(BROWSER) htmlcov/index.html
 
-docs:
-	rm -f docs/xncml.rst
-	rm -f docs/modules.rst
-	sphinx-apidoc -o docs/ xncml
-	$(MAKE) -C docs clean
+autodoc: clean-docs ## create sphinx-apidoc files:
+	sphinx-apidoc -o docs/apidoc --private --module-first xncml
+
+linkcheck: autodoc ## run checks over all external links found throughout the documentation
+	$(MAKE) -C docs linkcheck
+
+docs: autodoc ## generate Sphinx HTML documentation, including API docs
 	$(MAKE) -C docs html
-	open docs/_build/html/index.html
+ifndef READTHEDOCS
+	$(BROWSER) docs/_build/html/index.html
+endif
 
-release: clean
-	python setup.py sdist upload
-	python setup.py bdist_wheel upload
+servedocs: docs ## compile the docs watching for changes
+	watchmedo shell-command -p '*.rst' -c '$(MAKE) -C docs html' -R -D .
 
-sdist: clean
-	python setup.py sdist
-	python setup.py bdist_wheel upload
+dist: clean ## builds source and wheel package
+	python -m flit build
 	ls -l dist
+
+release: dist ## package and upload a release
+	python -m flit publish dist/*
+
+install: clean ## install the package to the active Python's site-packages
+	python -m flit install
```

Comment:
Shouldn't it do `pip install .` instead?
Otherwise, ensure `pip install flit` is ran before using `flit install`. 
A user has to manually install it, which goes against the point of using the makefile to simplify the workflow.